src/NonlinearArithmetic/Power.dfy

```diff
@@ -199,6 +199,26 @@ module Power {
     }
   }
 
+  lemma LemmaPowSubs(b: int, e1: nat, e2: nat)
+    decreases e1
+    requires e1 >= e2
+    ensures Pow(b, e1 - e2) * Pow(b, e2) == Pow(b, e1)
+  {
+    LemmaPowAdds(b, e1 - e2, e2);
+  }
+
+  lemma LemmaPowSubsAuto()
+    ensures forall b: int, e1: nat, e2: nat {:trigger Pow(b, e1 - e2)} | e1 >= e2
+      :: Pow(b, e1 - e2) * Pow(b, e2) == Pow(b, e1)
+  {
+    reveal Pow();
+    forall b: int, e1: nat, e2: nat {:trigger Pow(b, e1 - e2)} | e1 >= e2
+      ensures Pow(b, e1 - e2) * Pow(b, e2) == Pow(b, e1)
```

Comment:
You can remove this `ensures`

Reply:
Indeed I can! There are many other instances of this pattern in this file, but perhaps those could be addressed later. I'll simplify this one for now.